### ISSUE TITLE
waitConfirmations deploy options not waiting

### ISSUE THREAD

**Issue by PatrickAlphaC** [wighawag/hardhat-deploy#267]
**Describe the bug**
The `waitConfirmations`  `deployOptions` (extended from `TxOptions`) doesn't actually wait for the x number of blocks. 

**To Reproduce**
1. Have some simple contract code in contracts.
2. Create your `01-deploy.js` in your `deploy` folder, and give it a deploy script like this:
```
const fundMe = await deploy("FundMe", {
    from: deployer,
    args: [],
    log: true,
    waitConfirmations: 15,
  })
```

It'll then tell you the contract has been deployed, but if you check any block explorer right away, you'll clearly see 15 blocks have not been passed. This is an issue because if I want to programmatically verify my contracts, I need to wait x block confirmations before I send an API call to etherscan. 

**Expected behavior**
I expect it to wait the x block confirmations 

**versions**
 - hardhat-deploy `0.9.27`
 - hardhat `2.8.3`
 - nodejs `v14.17.0` & `v16.13.2`

**Comment by wighawag** [comment]
Thanks for the report

should be fixed in 0.9.29

Feel free to reopen if there is still an issue

**Comment by PatrickAlphaC** [comment]
Can confirm it works!
